### PR TITLE
Optimize url encoding, use default port for debugger service

### PR DESCRIPTION
**Pull request by annagrin**
- Optimize encoding of the extension url by not awaiting on the result until needed.
  - This cuts the startup time of dwds.
- Try binding debug service to a constant port if possible.
  - This will help with caching VM's API responses.

closes: https://github.com/dart-lang/webdev/issues/1400
helps: https://github.com/dart-lang/webdev/issues/1401

### CODE REVIEW

File: dwds/lib/dwds.dart [hunk 1]
```diff
@@ -134,13 +134,13 @@ class Dwds {
...
-      extensionUri = Uri(
+      extensionUri = Future.value(Uri(
               scheme: useSseForDebugBackend ? 'http' : 'ws',
               host: extensionBackend.hostname,
               port: extensionBackend.port,
               path: r'$debug')
-          .toString();
-      if (urlEncoder != null) extensionUri = await urlEncoder(extensionUri);
+          .toString());
+      if (urlEncoder != null) extensionUri = urlEncoder(await extensionUri);
```
Comment (grouma) [inline dwds/lib/dwds.dart:143]: What's the intention here? I'm not sure why this is beneficial?

Comment (annagrin) [reply dwds/lib/dwds.dart:143]: That makes is run the first urlEncoder in parallel with starting the app, before that we were blocking on it unnecessarily.
That speeds up the time from start of DDR and loading the app, and allows to delay the first urlEncoder call if needed (as it currently needed as some bug prevents us from connecting to the pen storage server before the build process starts running).

Comment (grouma) [reply dwds/lib/dwds.dart:143]: Oh I see now. the type of extensionUri has changed to `Future<Uri>`.

**Review by grouma — APPROVED** [review]

**Comment by annagrin** [comment]
I will wait with submitting this until we have metrics ready.

File: dwds/lib/src/services/debug_service.dart [hunk 2]
```diff
@@ -252,7 +252,7 @@ class DebugService {
         return innerHandler(request);
       };
     }
-    var server = await startHttpServer(hostname);
+    var server = await startHttpServer(hostname, port: 44456);
```
Comment (nshahan) [inline dwds/lib/src/services/debug_service.dart:255]: Is this port hardcoded anywhere else? Where did this value come from?

**Review by nshahan — APPROVED** [review]

File: dwds/lib/src/services/debug_service.dart (hunk 2 above)
Comment (annagrin) [reply dwds/lib/src/services/debug_service.dart:255]: Just random:) If it is not available, we will pick another one in `startHttpServer`.

**Comment by annagrin** [comment]
rebased onto latest master

**Pull request merged**